### ISSUE TITLE
Invalid callback id received by sendPluginResult (reopen)

### ISSUE THREAD

**Issue by thor-the-norseman** [dpa99c/cordova-diagnostic-plugin#62]
See issue #43, that I can't "reopen" as instructed, because I don't own the issue.

I installed your plugin via the Ionic framework (which supports Cordova plugins) and I'm trying to call `isLocationEnabled()`, and in the iOS console, it says `Invalid callback id received by sendPluginResult` and I never get a callback. My code runs inside an Angular controller "on enter" event handler and looks like this:

```
        cordova.plugins.diagnostic.isLocationEnabled(function(enabled) {
          console.log("Location enabled:", enabled);
        }, function() {
          console.log("Error"):
        });
```

My app is due to be launched tomorrow, and suddenly someone comes along and tells me that the app needs to bring people to the Location Settings screen when they have disabled their GPS, so I scramble around and find your plugin through Google, and here I am, unable to implement it.

**Comment by dpa99c** [comment]
The `Invalid callback id received by sendPluginResult` error message is being logged due a bug introduced in v2.3.14 of the plugin and has been resolved in commit 84c68ca0a830f52418aebcfdf5292b5d96e806ba which is published as v2.3.17 of the plugin. The bug is caused because the location authorization change event is being fired at app startup before a location request callback ID has been registered. However, the only side effect of this bug is that this log message appears at app startup - it is unrelated to your issue of a callback for `isLocationEnabled()` not being successfully executed.

Try building and running this simple Cordova test project which implements a plugin call very similar to yours above: [test_project.zip](https://github.com/dpa99c/cordova-diagnostic-plugin/files/261192/test_project.zip)

The result should look something like this, which is a screenshot of it running on the iOS 9.3 Simulator:

![screen shot 2016-05-12 at 12 37 19](https://cloud.githubusercontent.com/assets/2345062/15213577/0bff6a5e-1840-11e6-90e7-83399ae0128c.png)

**Comment by dpa99c** [comment]
Reopen if still an issue
